torch/csrc/autograd/engine.h

```diff
@@ -64,6 +64,7 @@ struct TORCH_API Engine {
   std::once_flag start_threads_flag_;
   std::vector<std::shared_ptr<ReadyQueue>> ready_queues_;
   std::vector<std::function<void()>> final_callbacks_;
+  // To protect accesses to final_callbacks_
```

Comment:
Generic wording note: it's probably clearest if we say explicitly that the lock protects both reads and writes